engine/src/main/java/org/destinationsol/game/SaveManager.java

```diff
@@ -52,29 +76,100 @@ public static void writeShip(HullConfig hull, float money, ArrayList<SolItem> it
                 }
             }
         }
-        IniReader.write(FILE_NAME, "hull", hullName, "money", (int) money, "items", sb.toString());
+
+        return sb.toString();
+
+    }
+
+    /**
+     * Writes the player's mercenaries to their JSON file.
+     * Will create file if it doesn't exist.
+     * @param game The instance of the game we're dealing with
+     */
+    private static void writeMercs(SolGame game) {
+        PrintWriter writer;
+        
+        ItemContainer mercIc = game.getHero().getTradeContainer().getMercs();
+        
+        List<JsonObject> jsons = new ArrayList<JsonObject>();
+        JsonObject json;
+        
+        for (List<SolItem> group : mercIc) {
+            for (SolItem item : group) {
+                SolShip merc = ((MercItem) item).getSolShip();
+                // Json fields
+                String hullName = game.getHullConfigs().getName(merc.getHull().config);
+                int money = (int) merc.getMoney();
+                
+                ArrayList<SolItem> itemsList = new ArrayList<SolItem>();
+                for (List<SolItem> group1 : merc.getItemContainer()) {
+                    for (SolItem i : group1) {
+                        itemsList.add(0, i);
+                    }
+                }
+                String items = itemsToString(itemsList);
+                
+                json = new JsonObject();
```

Comment:
conscience

Reply:
Always try to reduce the scope of your variables as much as you can.

Reply:
Ok, that makes sense.